packages/react-server/src/ReactFlightServer.js

```diff
@@ -669,6 +675,16 @@ export function resolveModelToJSON(
   key: string,
   value: ReactClientValue,
 ): ReactJSONValue {
+  if (typeof value === 'string' && value[value.length - 1] === 'Z') {
+    // Possibly a Date, whose toJSON automatically calls toISOString
+    // $FlowFixMe[incompatible-use]
+    const originalValue = parent[key];
+    // $FlowFixMe[method-unbinding]
+    if (Object.prototype.toString.call(originalValue) === '[object Date]') {
```

Comment:
Probably need to decide on how much we care about cross-realm vs perf and pick the same strategy for FormData and other future values. There will be quite a few. If we go with toString, it might be worth switching on the value.

We could also check isArray/instanceof first and only fallback to the toString for rarer objects.

Reply:
Changed to instanceof for now to match yours.